components/list/list-item-mixin.js

```diff
@@ -0,0 +1,268 @@
+				flex-grow: 0;
+				flex-shrink: 0;
+				margin: 0.6rem 0.9rem 0.6rem 0;
+			}
+
+			:host([dir="rtl"]) input[type="checkbox"].d2l-input-checkbox {
+				margin-left: 0.9rem;
+				margin-right: 0;
+			}
+
+			.d2l-list-item-main {
+				flex-grow: 1;
+				margin-top: 0.05rem;
+			}
+
+			.d2l-list-item-content-flex ::slotted([slot="actions"]) {
+				align-self: flex-start;
+				display: grid;
+				flex-grow: 0;
+				grid-auto-columns: 1fr;
+				grid-auto-flow: column;
+				grid-gap: 0.3rem;
+				margin: 0.15rem 0;
+				z-index: 4;
+			}
+
+			a, label {
+				height: 100%;
+				position: absolute;
+				width: 100%;
+				z-index: 2;
+			}
+
+			:host([href]) label {
+				width: 2.1rem;
+				z-index: 3;
+			}
+
+			:host([href]) {
+				--d2l-list-item-content-text-color: var(--d2l-color-celestine);
+			}
+
+			a[href]:focus + .d2l-list-item-content,
+			a[href]:hover + .d2l-list-item-content {
+				--d2l-list-item-content-text-decoration: underline;
+			}
+
+			:host([href]) .d2l-list-item-link:focus {
+				outline: none;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this._breakpoint = 0;
+		this.breakpoints = defaultBreakpoints;
+		this.role = 'listitem';
+		this.selectable = false;
+		this._contentId = getUniqueId();
+		this._checkBoxId = getUniqueId();
+	}
+
+	get breakpoints() {
+		return this._breakpoints;
+	}
+
+	set breakpoints(value) {
+		const oldValue = this._breakpoints;
+		if (value !== defaultBreakpoints) this._breakpoints = value.sort((a, b) => b - a).slice(0, 4);
+		else this._breakpoints = defaultBreakpoints;
+		this.requestUpdate('breakpoints', oldValue);
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+
+		const separators = this.parentNode.getAttribute('separators');
+		if (separators) this._separators = separators;
+		this._extendSeparators = this.parentNode.hasAttribute('extend-separators');
+
+		ro.observe(this);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		ro.unobserve(this);
+	}
+
+	focus() {
+		const node = getFirstFocusableDescendant(this);
+		if (node) node.focus();
+	}
+
+	render() {
+		const label = this.selectable ? html`<label class="d2l-list-item-label" for="${this._checkBoxId}" aria-labelledby="${this._contentId}"></label>` : null;
+		const link = this.href ? html`<a class="d2l-list-item-link" href="${ifDefined(this.href)}" aria-labelledby="${this._contentId}"></a>` : null;
+
+		const classes = {
+			'd2l-list-item-container': true,
+			'd2l-list-item-flex': label || link || !!this._renderOutside(),
+			'd2l-visible-on-ancestor-target': true
+		};
+
+		return html`
+			<div class="${classMap(classes)}" breakpoint="${this._breakpoint}">
+				${label}
+				${this._renderOutside()}
+				${link}
+				<div id="${this._contentId}"
+					class="d2l-list-item-content"
+					?extend-separators="${this._extendSeparators}"
+					separators="${ifDefined(this._separators)}">
+					<div class="d2l-list-item-content-flex">
+						${this._renderBeforeContent()}
```

Comment:
I decided to add a `renderBeforeContent()` since this might be changed more than just `renderContent()`? It could also be just more useful for my test case to have it.